azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -0,0 +1,360 @@
+                "entityPath": entity_path,
+                "identity": endpoint_identity
+            })
+        elif EndpointType.CosmosDBCollection.value == endpoint_type.lower():
+            if authentication_type != AuthenticationType.IdentityBased.value and not connection_string:
+                # try to get connection string
+                get_cosmos_db_cstring(
+                    cmd=self.cmd,
+                    account_name=endpoint_account_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            if connection_string:
+                # parse out key from connection string
+                if not primary_key and not secondary_key:
+                    parsed_cs = parse_cosmos_db_connection_string(connection_string)
+                    primary_key = parsed_cs["AccountKey"]
+                    secondary_key = parsed_cs["AccountKey"]
+                # parse out endpoint uri from connection string
+                if not endpoint_uri:
+                    endpoint_uri = parsed_cs["AccountEndpoint"]
+            if authentication_type != AuthenticationType.IdentityBased.value and not any([primary_key, secondary_key]):
+                raise RequiredArgumentMissingError(
+                    "Primary key via --primary-key, secondary key via --secondary-key, or connection string via "
+                    "--connection-string is required."
+                )
+            if primary_key and not secondary_key:
+                secondary_key = primary_key
+            if secondary_key and not primary_key:
+                primary_key = secondary_key
+            if not endpoint_uri:
+                raise RequiredArgumentMissingError(
+                    "Endpoint uri via --endpoint-uri or connection string via --connection-string is required."
+                )
+            if partition_key_name and not partition_key_template:
+                partition_key_template = '{deviceid}-{YYYY}-{MM}'
+            endpoints.cosmos_db_sql_collections.append({
+                "name": endpoint_name,
+                "databaseName": database_name,
+                "collectionName": collection_name,
+                "primaryKey": primary_key,
+                "secondaryKey": secondary_key,
+                "partitionKeyName": partition_key_name,
+                "partitionKeyTemplate": partition_key_template,
+                "subscriptionId": endpoint_subscription_id,
+                "resourceGroup": endpoint_resource_group,
+                "authenticationType": authentication_type,
+                "endpointUri": endpoint_uri,
+                "identity": endpoint_identity
+            })
+        elif EndpointType.AzureStorageContainer.value == endpoint_type.lower():
+            if authentication_type != AuthenticationType.IdentityBased.value and not connection_string:
+                # try to get connection string
+                get_storage_cstring(
```

Comment:
Same question here - if the calls are just for validation or if we want to use the returned value:
```suggestion
                connection_string = get_storage_cstring(
```